### PR TITLE
Added new file for respectful code reviews

### PR DESCRIPTION
**Pull request by ShvetaVasisht**
Closes #[317]

## If an article related PR

- [x] I have checked that my suggestion follows the Contribution guide

- [x]  I have added which issue this PR closes

- [x] I have 0 warnings from Danger.js

- [x] If I have added images, they are in /files/ folder

## If other PR

- [ ] I have added which issue this PR closes

### CODE REVIEW

**Comment by prokopsimek** [comment]
@ShvetaVasisht Thank you for your pull request! The article is great but could you minimize the plagiarism and write it in your own words?

image

Thanks! 🙏

**Comment by ShvetaVasisht** [comment]
Please let me know if I should create a new pull request thereafter

**Comment by prokopsimek** [comment]
@ShvetaVasisht Awesome, thank you! 👏

image

@ridlees May I ask you for a review?

**Review by prokopsimek — APPROVED** [review]

File: practices/respectful_code_reviews.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+Wherever possible, support your comments with an explanation. Suggestions and examples help the developer to understand the comments better. If you are suggesting alternate solutions highlight the advantages.
+
+**Be respectful and clear**
+
+Use clear and respectful language. The development and review process aims to improve the quality of the product hence make comments keeping the code as the main subject of the discussion.
+
+**Appreciate good work during code review**
+
+Recognize good work whenever you see it. Acknowledging good work boosts morale of the developer and lays the foundation for a healthy work environment.
```
Comment (ridlees) [inline practices/respectful_code_reviews.md:67]:
```suggestion
Recognize good work whenever you see it. Acknowledging good work boosts the morale of the developer and lays the foundation for a healthy work environment.
```

File: practices/respectful_code_reviews.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+
+
+## Problems Respectful Code Reviews Helps to Solve
+
+- [Toxic team culture](/problems/toxic-team-culture)
+- Knowledge Stagnation  e.g. open discussions lead to knowledge sharing.
+- [Long feedback loops](/problems/long-feedback-loops)
+- [Poor code quality](/problems/poor-code-quality)
+
```
Comment (ridlees) [inline practices/respectful_code_reviews.md:46]:
```suggestion
- [Finger-Pointing And Blaming](/problems/finger-pointing-and-blaming)
```


I would see the blaming problem as linked. What do you think?

File: practices/respectful_code_reviews.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+# Respectful Code Reviews
+
+**TL;DR**
+
+Respectful and constructive feedback is the basis of a fruitful collaboration, healthy work environment, and quality product.
+
+
+## Why You Might Want Respectful Code Reviews
+
+Code review helps to improve the quality of software projects but negative and unclear code review comments can have unwanted consequences.
```
Comment (ridlees) [inline practices/respectful_code_reviews.md:35]:
```suggestion
Code review helps you to improve the quality of software projects. However, negative and unclear code review comments can have unwanted consequences.
```

File: practices/respectful_code_reviews.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+The right style of communication not only helps to build relationships and healthy work environments but also helps to achieve the final goal of delivering a quality product. It improves the overall health of the system and code.
```
Comment (ridlees) [inline practices/respectful_code_reviews.md:36]:
```suggestion
The right style of communication helps to build developer relationships and healthy work environments.  Respectful Code Reviews help to achieve the final goal of delivering a quality product. They improve the overall health of the system and code.```

File: practices/respectful_code_reviews.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+Define guidelines at the project or company level. Follow style guides and coding standard documents and use them as benchmarks for reviews. These guidelines should be available to all concerned for reference.
+
+**Understand code and functionality before reviewing**.
+
+Have a clear picture of the functionality before you start a review. Read related documentation. You can even ask for a code walkthrough. Clarify doubts, if any.
+
+**Explain review comments**
+
+Wherever possible, support your comments with an explanation. Suggestions and examples help the developer to understand the comments better. If you are suggesting alternate solutions highlight the advantages.
```
Comment (ridlees) [inline practices/respectful_code_reviews.md:59]:
```suggestion
Wherever possible, support your comments with an explanation. Suggestions and examples help the developer to understand the comments better. If you are proposing alternative solutions, show the advantages.
```

File: practices/respectful_code_reviews.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+Recognize good work whenever you see it. Acknowledging good work boosts morale of the developer and lays the foundation for a healthy work environment.
+
+ **Try to understand the intent of feedback**
+
+As a developer, you should read the comments keeping in mind that review comments are quality checks for the code. They help to make the code and the product better.
+
+ **Common Pitfalls**
+
+-  Written communication makes it hard to understand the reasoning behind the comment especially if the developer and reviewer are not acquainted.
```
Comment (ridlees) [inline practices/respectful_code_reviews.md:75]:
```suggestion
-  Written communication makes it hard to understand the comment's intent, especially if the developer and reviewer don't know each other.
```

File: practices/respectful_code_reviews.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+Have a clear picture of the functionality before you start a review. Read related documentation. You can even ask for a code walkthrough. Clarify doubts, if any.
+
+**Explain review comments**
+
+Wherever possible, support your comments with an explanation. Suggestions and examples help the developer to understand the comments better. If you are suggesting alternate solutions highlight the advantages.
+
+**Be respectful and clear**
+
+Use clear and respectful language. The development and review process aims to improve the quality of the product hence make comments keeping the code as the main subject of the discussion.
```
Comment (ridlees) [inline practices/respectful_code_reviews.md:63]:
```suggestion
Use clear and respectful language. The development and review process aims to improve the quality of the product. Make your code the main subject of the review discussion.
```

**Review by ridlees — CHANGES_REQUESTED** [review]
Hello @ShvetaVasisht, thanks for the article and great writing. I like it. 

Please, you don't need to do another PR.

Our articles always contain some "cover" image. You can use photobank (like Unsplash) or something you see more fitting. 

I suggested some minor changes to keep your article with DX Knowledgebase style.

File: practices/respectful_code_reviews.md (hunk 4 above)
Comment (ShvetaVasisht) [reply practices/respectful_code_reviews.md:36]: They improve the overall health of the system and code.```
I think extra characters have been added in the suggestion at the end

File: practices/respectful_code_reviews.md (hunk 1 above)
Comment (ShvetaVasisht) [reply practices/respectful_code_reviews.md]: Suggestion applied

File: practices/respectful_code_reviews.md (hunk 4 above)
Comment (ShvetaVasisht) [reply practices/respectful_code_reviews.md:36]: Also, I meant  by relationships I meant workplace relationships between reviewer and developer(both may or may not be developers). Maybe, we can just keep the following point -The right style of communication helps to build  healthy work environments.  Respectful Code Reviews help to achieve the final goal of delivering a quality product. They improve the overall health of the system and code. 
If this helps clear confusion

Comment (ShvetaVasisht) [reply practices/respectful_code_reviews.md:36]: Also, by relationships I meant workplace relationships between reviewer and developer(both may or may not be developers). Maybe, we can just keep the following point -The right style of communication helps to build healthy work environments. Respectful Code Reviews help to achieve the final goal of delivering a quality product. They improve the overall health of the system and code.
If this helps clear confusion

File: practices/respectful_code_reviews.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+Respectful and constructive feedback is the basis of a fruitful collaboration, healthy work environment, and quality product.
+
+
+## Why You Might Want Respectful Code Reviews
+
+Code review helps to improve the quality of software projects but negative and unclear code review comments can have unwanted consequences.
+The right style of communication not only helps to build relationships and healthy work environments but also helps to achieve the final goal of delivering a quality product. It improves the overall health of the system and code.
+
+
```
[the remark this answers is not in the document]
Comment (ShvetaVasisht) [reply practices/respectful_code_reviews.md]: added image

[the remark this answers is not in the document]
Comment (ridlees) [reply practices/respectful_code_reviews.md]: Hello, sorry for the late reply. Can you please add it to the article using the following syntax? 

Alt Name [Name](https://image source)

File: practices/respectful_code_reviews.md (hunk 4 above)
Comment (ridlees) [reply practices/respectful_code_reviews.md:36]: I see, use either your first version or the second (fact that I got confused doesn't mean it is not clear for 99 % of people)

File: practices/respectful_code_reviews.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (ShvetaVasisht) [reply practices/respectful_code_reviews.md:37]: I have created the image so there is no image source. I have added the image name

File: practices/respectful_code_reviews.md (hunk 4 above)
Comment (ShvetaVasisht) [reply practices/respectful_code_reviews.md]: ok

File: practices/respectful_code_reviews.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (ShvetaVasisht) [reply practices/respectful_code_reviews.md:37]: Added to article Respectful Code Review.The jpg is also added to files folder

**Review by ridlees — APPROVED** [review]
Thanks for the work @ShvetaVasisht! 🎉 🎉

**Comment by ShvetaVasisht** [comment]
Thanks for your support and time @prokopsimek @ridlees

**Comment by prokopsimek** [comment]
@ShvetaVasisht You're welcome! Thank you for your PR! 👏

**Pull request merged**